qa/tasks/workunit.py

```diff
@@ -418,6 +418,6 @@ def _run_tests(ctx, refspec, role, tests, env, basedir,
         remote.run(
             logger=log.getChild(role),
             args=[
-                'rm', '-rf', '--', workunits_file, clonedir,
+                'rm', '-rf', '--', workunits_file, clonedir, scratch_tmp
```

Comment:
Odd - I thought I left a review for this, but I don't see it. 

I had mentioned that a few lines above, ```scratch_tmp``` is getting conditionally cleaned if ```cleanup``` is true. Should that section be removed now that it's always deleting the scratch directory?

Reply:
Good point. I think we should keep that check. I moved the scratch cleanup to the finalizer and conditionally removed `scratch_tmp`.